File: backend/app/schemas/call.py
from pydantic import BaseModel, Field


class TranscriptItem(BaseModel):
    speaker: str
    text: str
    timestamp: str


class SentimentItem(BaseModel):
    text: str
    sentiment: str
    confidence: float


class InitiateCallRequest(BaseModel):
    restaurant_name: str
    restaurant_phone: str
    party_size: int
    preferred_date: str
    preferred_time: str
    buffer_minutes: int = 30
    dietary_restrictions: list[str] = Field(default_factory=list)
    user_name: str


class CallStatusResponse(BaseModel):
    call_id: str
    status: str
    transcript: list[TranscriptItem] = Field(default_factory=list)
    sentiment_log: list[SentimentItem] = Field(default_factory=list)
    confirmation_code: str | None = None
    confirmed_time: str | None = None
    confirmed_date: str | None = None


class InviteAttendee(BaseModel):
    name: str
    email: str


class GenerateICSRequest(BaseModel):
    call_id: str
    attendees: list[InviteAttendee] = Field(default_factory=list)
    restaurant_name: str
    restaurant_address: str
    confirmed_date: str
    confirmed_time: str
    restaurant_phone: str
    dietary_notes: str


class SendInvitesRequest(BaseModel):
    ics_content: str
    attendees: list[InviteAttendee] = Field(default_factory=list)
    restaurant_name: str
    confirmed_date: str
    confirmed_time: str

File: frontend/types/call.ts
export type CallStatus =
  | 'initiated'
  | 'ringing'
  | 'connected'
  | 'negotiating'
  | 'confirmed'
  | 'completed'
  | 'busy'
  | 'no_answer'
  | 'canceled'
  | 'failed';

export type TranscriptLine = {
  speaker: string;
  text: string;
  timestamp: string;
};

export type SentimentEntry = {
  text: string;
  sentiment: string;
  confidence: number;
};

export type ProgressStep = {
  label: string;
  completed: boolean;
  active: boolean;
};

export type CallRecord = {
  id: string;
  booking_id: string;
  restaurant_name: string;
  restaurant_phone: string;
  status: CallStatus;
  transcript: TranscriptLine[];
  sentiment_log: SentimentEntry[];
  confirmation_code?: string | null;
  confirmed_time?: string | null;
  confirmed_date?: string | null;
  created_at: string;
  duration_seconds?: number | null;
};

export type InitiateCallRequest = {
  restaurant_name: string;
  restaurant_phone: string;
  party_size: number;
  preferred_date: string;
  preferred_time: string;
  buffer_minutes?: number;
  dietary_restrictions: string[];
  user_name: string;
  conversation_style?: 'strict' | 'friendly';
};

export type CallStatusResponse = {
  call_id: string;
  status: string;
  transcript: TranscriptLine[];
  sentiment_log: SentimentEntry[];
  confirmation_code?: string | null;
  confirmed_time?: string | null;
  confirmed_date?: string | null;
};

export type CallControlRequest = {
  action: 'stop' | 'speak';
  message?: string;
};

export type InviteAttendee = {
  name: string;
  email: string;
};

export type CalendarInvite = {
  id: string;
  call_record_id: string;
  attendees: Array<InviteAttendee & { dietary?: string | null }>;
  ics_content: string;
  sent: boolean;
  sent_at?: string | null;
};

export type GenerateICSRequest = {
  call_id: string;
  attendees: InviteAttendee[];
  restaurant_name: string;
  restaurant_address: string;
  confirmed_date: string;
  confirmed_time: string;
  restaurant_phone: string;
  dietary_notes: string;
};

export type SendInvitesRequest = {
  ics_content: string;
  attendees: InviteAttendee[];
  restaurant_name: string;
  confirmed_date: string;
  confirmed_time: string;
};

export type CallStreamUpdate = {
  event: string;
  call_id: string;
  [key: string]: unknown;
};

export type SendInvitesResponse = {
  subject: string;
  results: Array<{
    name: string;
    email: string;
    success: boolean;
    error?: string;
  }>;
  success_count: number;
  failure_count: number;
};
